mongofiles/mongofiles.go

```diff
@@ -234,49 +249,38 @@ func (mf *MongoFiles) getTargetGFSFiles() ([]*gfsFile, error) {
 	var gridFiles []*gfsFile
 	var err error
 
+	var query bson.M
+
 	// If mongofiles get ... is called, then query for all files
 	// specified in mf.FileNameList -- otherwise, preserve correct
 	// behavior for mongofiles get_id ...
 	if len(mf.FileNameList) > 0 {
-		query := bson.M{"filename": bson.M{"$in": mf.FileNameList}}
-
-		gridFiles, err = mf.findGFSFiles(query)
+		query = bson.M{"filename": bson.M{"$in": mf.FileNameList}}
+	} else if mf.FileNameRegex != "" {
+		query = bson.M{
+			"filename": bson.M{
+				"$regex":   mf.FileNameRegex,
+				"$options": mf.StorageOptions.RegexOptions,
+			},
+		}
+	} else if mf.Id != "" {
+		id, err := mf.parseOrCreateID()
 		if err != nil {
 			return nil, err
 		}
 
-		if len(gridFiles) < len(mf.FileNameList) {
-			return nil, fmt.Errorf("requested files not found: %v", mf.FileNameList)
-		}
+		query = bson.M{"_id": id}
 	} else {
-		var queryProp string
-		var query string
-
-		if mf.Id != "" {
-			queryProp = "_id"
-			query = mf.Id
-
-			id, err := mf.parseOrCreateID()
-			if err != nil {
-				return nil, err
-			}
-			gridFiles, err = mf.findGFSFiles(bson.M{"_id": id})
-			if err != nil {
-				return nil, err
-			}
-		} else {
-			queryProp = "name"
-			query = mf.FileName
+		query = bson.M{"filename": mf.FileName}
```

Comment:
Does this case ever actually happen? I can't see any codepath that would get to this branch; `get_id` and `delete_id` will hit the `mf.Id != ""` branch, `get_regex` will hit the regex branch, and `get` will hit the `len(mf.FileNameList) > 0` branch.

Reply:
Yes, we need it for `mongofiles ... get ... --local ...`, it seems.